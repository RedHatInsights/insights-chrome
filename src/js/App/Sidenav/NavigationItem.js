import React from 'react';
import { NavItem } from '@patternfly/react-core4/dist/js/components/Nav/NavItem';
import PropTypes from 'prop-types';

const basepath = document.baseURI;

const NavigationItem = ({ itemID, title, parent, navigate, ...props }) => (
    <NavItem
        {...props}
        itemId={itemID}
        preventDefault
        to={navigate || `${basepath}${parent}/${itemID}`}
    >
        {title}
    </NavItem>
);

NavigationItem.propTypes = {
    itemID: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    title: PropTypes.node,
    navigate: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    parent: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};

NavigationItem.defaultProps = {
    parent: ''
};

export default NavigationItem;
