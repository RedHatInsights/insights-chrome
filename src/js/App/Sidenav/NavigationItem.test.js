import React from 'react';
import NavigationItem  from './NavigationItem';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('NavigationItem', () => {
    it('should render coorectly', () =>{
        const wrapper = shallow(<NavigationItem/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('parent with value and itemID undefined', () => {
        let props = { parent: 'someValue' };
        const wrapper = shallow(<NavigationItem {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('parent and itemID with value', () => {
        let props = { parent: 'someValue', itemID: 'someID' };
        const wrapper = shallow(<NavigationItem {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('parent undefined and itemID with value', () => {
        let props = { itemID: 'someID' };
        const wrapper = shallow(<NavigationItem {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('with some navigate value set', () => {
        let navigate = {};
        const wrapper = shallow(<NavigationItem navigate={navigate}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
