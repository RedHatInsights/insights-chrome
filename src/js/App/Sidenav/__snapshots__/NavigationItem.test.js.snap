// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem parent and itemID with value 1`] = `
<Component
  itemId="someID"
  preventDefault={true}
  to="http://localhost/someValue/someID"
/>
`;

exports[`NavigationItem parent undefined and itemID with value 1`] = `
<Component
  itemId="someID"
  preventDefault={true}
  to="http://localhost//someID"
/>
`;

exports[`NavigationItem parent with value and itemID undefined 1`] = `
<Component
  preventDefault={true}
  to="http://localhost/someValue/undefined"
/>
`;

exports[`NavigationItem should render coorectly 1`] = `
<Component
  preventDefault={true}
  to="http://localhost//undefined"
/>
`;

exports[`NavigationItem with some navigate value set 1`] = `
<Component
  preventDefault={true}
  to={Object {}}
/>
`;
